Jint/Runtime/Interpreter/Expressions/JintTemplateLiteralExpression.cs

```diff
@@ -0,0 +1,57 @@
+using System.Text;
+using Esprima.Ast;
+using Jint.Native;
+using Jint.Pooling;
+
+namespace Jint.Runtime.Interpreter.Expressions
+{
+    internal sealed class JintTemplateLiteralExpression : JintExpression
+    {
+        internal readonly TemplateLiteral _templateLiteralExpression;
+        internal JintExpression[] _expressions;
+
+        public JintTemplateLiteralExpression(Engine engine, TemplateLiteral expression) : base(engine, expression)
+        {
+            _templateLiteralExpression = expression;
+            _initialized = false;
+        }
+
+        protected override void Initialize()
+        {
+            DoInitialize();
+        }
+
+        internal void DoInitialize()
+        {
+            _expressions = new JintExpression[_templateLiteralExpression.Expressions.Count];
+            for (var i = 0; i < _templateLiteralExpression.Expressions.Count; i++)
+            {
+                var exp = _templateLiteralExpression.Expressions[i];
+                _expressions[i] = Build(_engine, exp);
+            }
+
+            _initialized = true;
+        }
+
+        private JsString BuildString()
+        {
+            var sb = StringBuilderPool.GetInstance();
+            for (var i = 0; i < _templateLiteralExpression.Quasis.Count; i++)
```

Comment:
Is there a number of fragments under which a concat is faster than using a builder?

Reply:
Tried concatenation (+=) against builder and the are pretty much same speed-wise. After 2 placeholders string builder's allocation profile gets better.